addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -76,6 +76,7 @@ const Wysiwyg = Widget.extend({
         this.$editable.data('oe-id', options.recordInfo.res_id);
         document.addEventListener('mousedown', this._onDocumentMousedown, true);
         this.$editable.on('blur', this._onBlur);
+        $('.o_not_editable').attr('contentEditable', false);
```

Comment:
Details but: the attribute is `contenteditable` not `contentEditable` (even if I guess browsers make it work with camelCase too), and I would always give string as value for the `attr` method (so `'false'`).

More important though: I would not search in the whole DOM if possible (even if it was like that before), but only in the DOM that the editor is configured to edit.

cc @dmo-odoo Is this ok to you?

Reply:
Thanks for notifying me ! :)

I would rather modify the `isNotEditableNode` function from `utils` to understand that `o_not_editable` is a magic class that makes the node not editable, just like we do with `oe_unremovable` and `oe_unbreakable`. However, I suspect this would end up having weird effects since the browser would not be aware of it, like, you would probably be able to edit in there for example, which would defeat the purpose of this change I suppose.

I guess we don't really have a choice in the matter, we *must* use the `contenteditable` attribute if we have any hope to make the browser understand what we are doing. However, I suggest to add this to the `_parse` function of `sanitize`. The function is poorly named as you might think it is only called once when loading the content in the editor but as far as I remember that is not the case. The sanitizing mechanism actually parses the nodes for every change so putting the logic there should even allow us to add support for dynamically toggling this magic `o_not_editable` class.

Therefore, I suggest to handle both the adding and removing of `o_not_editable` in the `_parse` function of `sanitize`, and I guess adding some tests for this particular behavior wouldn't hurt either... ;-)

Reply:
@detrouxdev I let you investigate, however:
- I don't think there is a "adding and removing" behavior for o_not_editable, it should never be removed but saved with the DOM as it is meant to indicate the node will never be editable (maybe I did not understand something? In that case, I'll let @detrouxdev investigate)

But indeed, if o_not_editable allows to put contenteditable="false" on dynamic content during edition, it's of course better.

Reply:
I guess it's somewhat acceptable to only handle the adding case in `_parse` and not the removing one if we really have no use for it.

Reply:
@dmo-odoo Can I ask that your team takes this fix about non editable parts and their related `contenteditable` attribute in a separate PR? It's becoming urgent as the website team already created 3-4 tasks about it with multiple attempts to solve the problem each time differently and each time it does not feel perfect :grimacing: (and meanwhile people are breaking page sections they are not supposed to be able to touch :grimacing:) And now that I was about to try and I understand what is being said here (and elsewhere), I see you merged another system to keep the `contenteditable` attribute on save 4 weeks ago with https://github.com/odoo/odoo/commit/5e3c466e1b6a23a76807ab4f4d81bcf70105eedb#diff-6bf09b72528757673aa5f73488708a44ba1af178d51205248e9bb8429d3e6584R2402 (if I understand correctly).

The main usecase that I knew was that `o_not_editable` was a class which allowed a content to receive the `contenteditable="false"` attribute when entering edition (and all of those attributes disappeared on save). It feels strange that behavior can coexist with another system to keep `contenteditable="false"` attribute outside of edit mode IMO. But `o_not_editable` may be part of more things (not just text edition)... who knows :grimacing: 

Thank you :)

Reply:
@dmo-odoo More discussions around this problem https://github.com/odoo/odoo/pull/83116#discussion_r834361159

Reply:
Getting urgent indeed ^^